pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -247,10 +283,25 @@ func validateRuleWithOperations(ruleWithOperations *admissionregistration.RuleWi
 	return allErrors
 }
 
+// hasUnrecognizedVersions returns true of any of the webhooks has an unrecognized version in AdmissionReviewVersions.
+func hasUnrecognizedAdmissionReviewVersions(webhooks []admissionregistration.Webhook) bool {
+	unusedFieldPath := field.NewPath("webhooks")
+	for _, hook := range webhooks {
+		// only skip checking for recognized version if we had invalid versions
```

Comment:
calling a validation function which validates several things twice, with different params, then using different results to infer a particular condition is hard to follow.

would be simpler to just check for a supported version in each webhook directly:
```go
// hasAcceptedAdmissionReviewVersions returns true if all webhooks have at least one
// admission review version this apiserver accepts.
func hasAcceptedAdmissionReviewVersions(webhooks []admissionregistration.Webhook) bool {
  for _, hook := range webhooks {
    hasRecognizedVersion := false
    for _, version := range hook.AdmissionReviewVersions {
      if isAcceptedAdmissionReviewVersion(version) {
        hasRecognizedVersion = true
        break
      }
    }
    if !hasRecognizedVersion {
      return false
    }
  }
  return true
}
```

Reply:
Hmm, I just realized the validation may have been wrong anyway till now.  A webhook is valid if it has at least one recognized version, right? The validation right now requires all valid values. will fix.